File: go/pkg/types/droplet.go
package types

import "time"

type Droplet struct {
	ID               int               `json:"id"`
	Name             string            `json:"name"`
	Memory           int               `json:"memory"`
	Vcpus            int               `json:"vcpus"`
	Disk             int               `json:"disk"`
	Locked           bool              `json:"locked"`
	Status           string            `json:"status"`
	Kernel           *Kernel           `json:"kernel"`
	CreatedAt        time.Time         `json:"created_at"`
	Features         []string          `json:"features"`
	BackupIDs        []int             `json:"backup_ids"`
	SnapshotIDs      []int             `json:"snapshot_ids"`
	Image            *Image            `json:"image"`
	Size             *Size             `json:"size"`
	SizeSlug         string            `json:"size_slug"`
	Networks         *Networks         `json:"networks"`
	Region           *Region           `json:"region"`
	Tags             []string          `json:"tags"`
	VolumeIDs        []string          `json:"volume_ids"`
	VPCUUID          string            `json:"vpc_uuid"`
}

type Kernel struct {
	ID      int    `json:"id"`
	Name    string `json:"name"`
	Version string `json:"version"`
}

type Image struct {
	ID            int       `json:"id"`
	Name          string    `json:"name"`
	Type          string    `json:"type"`
	Distribution  string    `json:"distribution"`
	Slug          string    `json:"slug"`
	Public        bool      `json:"public"`
	Regions       []string  `json:"regions"`
	MinDiskSize   int       `json:"min_disk_size"`
	SizeGigabytes float64   `json:"size_gigabytes"`
	CreatedAt     time.Time `json:"created_at"`
	Description   string    `json:"description"`
	Tags          []string  `json:"tags"`
	Status        string    `json:"status"`
	ErrorMessage  string    `json:"error_message"`
}

type Networks struct {
	V4 []NetworkV4 `json:"v4"`
	V6 []NetworkV6 `json:"v6"`
}

type NetworkV4 struct {
	IPAddress string `json:"ip_address"`
	Netmask   string `json:"netmask"`
	Gateway   string `json:"gateway"`
	Type      string `json:"type"`
}

type NetworkV6 struct {
	IPAddress string `json:"ip_address"`
	Netmask   int    `json:"netmask"`
	Gateway   string `json:"gateway"`
	Type      string `json:"type"`
}

type CreateDropletRequest struct {
	Name              string   `json:"name"`
	Region            string   `json:"region"`
	Size              string   `json:"size"`
	Image             string   `json:"image"`
	SSHKeys           []string `json:"ssh_keys,omitempty"`
	Backups           bool     `json:"backups,omitempty"`
	IPv6              bool     `json:"ipv6,omitempty"`
	Monitoring        bool     `json:"monitoring,omitempty"`
	Tags              []string `json:"tags,omitempty"`
	UserData          string   `json:"user_data,omitempty"`
	PrivateNetworking bool     `json:"private_networking,omitempty"`
	Volumes           []string `json:"volumes,omitempty"`
	VPCUUID           string   `json:"vpc_uuid,omitempty"`
}

type DropletsResponse struct {
	Droplets []Droplet `json:"droplets"`
	Links    Links     `json:"links"`
	Meta     Meta      `json:"meta"`
}

type DropletResponse struct {
	Droplet Droplet `json:"droplet"`
	Links   Links   `json:"links,omitempty"`
	Meta    Meta    `json:"meta,omitempty"`
}

type CreateDropletResponse struct {
	Droplet Droplet   `json:"droplet"`
	Links   Links     `json:"links,omitempty"`
	Meta    Meta      `json:"meta,omitempty"`
	Actions []Action  `json:"actions,omitempty"`
}

File: rust/crates/do-types/src/droplet.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};
use super::{Region, Size, Links, Meta, Action};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Droplet {
    pub id: u32,
    pub name: String,
    pub memory: u32,
    pub vcpus: u32,
    pub disk: u32,
    pub locked: bool,
    pub status: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub kernel: Option<Kernel>,
    pub created_at: DateTime<Utc>,
    pub features: Vec<String>,
    pub backup_ids: Vec<u32>,
    pub snapshot_ids: Vec<u32>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub image: Option<Image>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub size: Option<Size>,
    pub size_slug: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub networks: Option<Networks>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub region: Option<Region>,
    pub tags: Vec<String>,
    pub volume_ids: Vec<String>,
    pub vpc_uuid: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Kernel {
    pub id: u32,
    pub name: String,
    pub version: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Image {
    pub id: u32,
    pub name: String,
    #[serde(rename = "type")]
    pub image_type: String,
    pub distribution: String,
    pub slug: String,
    pub public: bool,
    pub regions: Vec<String>,
    pub min_disk_size: u32,
    pub size_gigabytes: f64,
    pub created_at: DateTime<Utc>,
    pub description: String,
    pub tags: Vec<String>,
    pub status: String,
    pub error_message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Networks {
    pub v4: Vec<NetworkV4>,
    pub v6: Vec<NetworkV6>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NetworkV4 {
    pub ip_address: String,
    pub netmask: String,
    pub gateway: String,
    #[serde(rename = "type")]
    pub network_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NetworkV6 {
    pub ip_address: String,
    pub netmask: u32,
    pub gateway: String,
    #[serde(rename = "type")]
    pub network_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDropletRequest {
    pub name: String,
    pub region: String,
    pub size: String,
    pub image: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub ssh_keys: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub backups: Option<bool>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub ipv6: Option<bool>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub monitoring: Option<bool>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub user_data: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub private_networking: Option<bool>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub volumes: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub vpc_uuid: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DropletsResponse {
    pub droplets: Vec<Droplet>,
    pub links: Links,
    pub meta: Meta,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DropletResponse {
    pub droplet: Droplet,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub links: Option<Links>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub meta: Option<Meta>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDropletResponse {
    pub droplet: Droplet,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub links: Option<Links>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub meta: Option<Meta>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub actions: Option<Vec<Action>>,
}